react/FieldMessage/FieldMessage.js

```diff
@@ -35,27 +38,25 @@ export default class FieldMessage extends Component {
     }
   };
 
-  constructor() {
-    super();
-
-    this.renderMessage = this.renderMessage.bind(this);
-    this.renderMessageIcon = this.renderMessageIcon.bind(this);
-  }
-
-  renderMessage() {
-    const { id, message, valid } = this.props;
+  renderMessage = () => {
+    const { id, message, tone, valid } = this.props;
 
     if (message) {
       const { critical, positive, secondary, ...restMessageProps } = this.props.messageProps;
 
+      const toneDefined = typeof tone !== 'undefined';
+      const showCritical = toneDefined ? tone === TONE.CRITICAL : valid === false;
+      const showPositive = toneDefined ? tone === TONE.POSITIVE : valid === true;
+      const showSecondary = tone === TONE.NEUTRAL || typeof valid === 'undefined';
+
       return (
         <Text
           id={id}
           small
           {...restMessageProps}
-          critical={(valid === false && !secondary) || critical}
-          positive={(valid === true && !secondary) || positive}
-          secondary={typeof valid === 'undefined' || secondary}
+          critical={(showCritical && !secondary) || critical}
```

Comment:
Suggestions from @conshow - could `critical`, `positive` and `secondary` override both tone and valid here to reduce complexity?